numpy/core/src/multiarray/getset.c

```diff
@@ -461,14 +460,13 @@ array_descr_set(PyArrayObject *self, PyObject *arg)
     if (_may_have_objects(PyArray_DESCR(self)) || _may_have_objects(newtype)) {
         npy_cache_import("numpy.core._internal", "_view_is_safe", &checkfunc);
         if (checkfunc == NULL) {
-            return -1;
+            goto fail;
```

Comment:
Nice catch. Have to say I continue to dislike `if`-statements with side effects... (obviously outside of this PR, though)